devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy

```diff
@@ -61,12 +68,31 @@ def updated() {
 	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
 }
 
+def getCommandClassVersions() {
+	[
+			0x71: alarmVersion, // Alarm
+			0x72: 1, // Manufacturer Specific
+			0x80: 1, // Battery
+			0x84: 1, // Wake Up
+	]
+}
+
+def getAlarmVersion() {
+	def alarmVersion = null
+	switch (getDataValue("MSR")) {
+		case "026F-0001-0001": alarmVersion = 3 //FireAngel ZST-630
+			break
+		default: alarmVersion = 2
+	}
+	return alarmVersion
```

Comment:
v3 is backwards compatible to v2. You shouldn't be checking the msr to determine the version, just integrate the parsing for First Alert into the v3 handler.

Reply:
I will merge some code from AlarmReport  to NotificationReport  (of course including field name changes due Notification repacement of Alarm )